vNext/src/Microsoft.AspNet.OData/Builder/ODataModelBuilder.cs

```diff
@@ -605,7 +605,7 @@ public virtual void ValidateModel(IEdmModel model)
 
             foreach (IEdmEntityType entity in model.SchemaElementsAcrossModels().OfType<IEdmEntityType>())
             {
-                if (!entity.IsAbstract && !entity.Key().Any())
+                if (!entity.IsAbstract && !entity.Key().Any() && entity.DeclaredProperties.Any())
```

Comment:
This was put in place to support primitive types like System.Boolean or System.Int32. The entity type mapping indicates that they are abstract but there are no keys available. This might be a bug somewhere else in the code.